src/main/scala/scalaz/stream/async/mutable/Queue.scala

```diff
@@ -45,10 +45,24 @@ trait Queue[A] {
    *
    * Please use `Topic` instead of `Queue` when all subscribers
    * need to see each value enqueued.
+   *
+   * This process is equivalent to `dequeueBatch(1)`.
    */
   def dequeue: Process[Task, A]
 
   /**
+   * Provides a process that dequeues in chunks.  Whenever *n* elements
+   * are available in the queue, `min(n, limit)` elements will be dequeud
+   * and produced as a single `Seq`.  Note that this naturally does not
+   * *guarantee* that `limit` items are returned in every chunk.  If only
+   * one element is available, that one element will be returned in its own
+   * sequence.  This method basically just allows a consumer to "catch up"
+   * to a rapidly filling queue in the case where some sort of batching logic
+   * is applicable.
+   */
+  def dequeueBatch(limit: Int = 0): Process[Task, Seq[A]]
```

Comment:
> `p1 ++ q.dequeueBatch(-1)`

Gotcha. Although in that case, the stack trace is going to be useless regardless of whether the error is thrown eagerly or not.

So, here's what I'd like to do:

* For `boundedQueue`, we already have `unboundedQueue`, so go ahead and make `boundQueue` parameter required, throw illegal arg exception eagerly.
* Same for `dequeueBatch`
* I'm fine with keeping `njoin` as is for now. Maybe we can revisit later.

> Personally, I actually feel that it's less ugly to define a subrange of Int to have a particular meaning, rather than making the function partial.

In this situation, I think more often than not, negative values are a bug (like the user flipped arguments in compute a limit, say). Obviously, this is somewhat subjective, but it's the decision I'd like to go with.